lib/awful/rules.lua

```diff
@@ -324,8 +325,16 @@ function rules.high_priority_properties.new_tag(c, value)
         -- Create a tag named after "value"
         t = atag.add(value, {screen=c.screen, volatile=true})
     elseif ty == "table" then
-        -- Assume a table of tags properties
-        t = atag.add(value.name or c.class or "N/A", value)
+        -- Assume a table of tags properties. Set the right screen, but
+        -- avoid editing the original table
+        local values = value.screen and value or util.table.clone(value)
+        values.screen = values.screen or c.screen
+
+        t = atag.add(value.name or c.class or "N/A", values)
+
+        -- In case the tag has been forced to another screen, move the client
+        c.screen = t.screen
+        props.screen = t.screen -- In case another rule query it
```

Comment:
> How about the following?

That's kind of the bug this PR try to fix. If the `new_tag` is `{name="Foo", screen=capi.mouse.screen}` using `awful.spawn`, then you don't want the default rule overrideing that. The spawn arguments always have priority over the default fallbacks (actually, this work because of a race condition, but whatever, works as long as `awful.spawn` is loaded after `awful.rules`).